bin/update-authorized-keys.rb

```diff
@@ -48,41 +55,55 @@ def build_authorized_keys
       authorized_keys += node["node"]["key"] + " @" + format(member["login"]) + "\n"
     end
   end
-
-  encoded_authorized_keys = Base64.strict_encode64(authorized_keys)
+  
+  authorized_keys  
 end
 
-def create_new_branch_commit(encoded_authorized_keys)
-  # Get the main branch sha
-  json = get_query(
-    token: ENV.fetch("GITHUB_TOKEN"),
-    api_url: GITHUB_REST_API_URL + "/git/ref/heads/main"
-  )
+def compare_with_existing_keys(authorized_keys)
 
-  sha = JSON.parse(json).dig("object", "sha")
+    # Get the latest blob for the content
+    api_url = GITHUB_REST_API_URL + "/contents/files/authorized_keys.txt"
 
-  # Create a reference which is basically creating a branch
-  json = post_query(
-    json: get_create_branch_query(sha),
-    token: ENV.fetch("GITHUB_TOKEN"),
-    api_url: GITHUB_REST_API_URL + "/git/refs"
-  )
+    json = get_query(
+      token: ENV.fetch("GITHUB_TOKEN"),
+      api_url: api_url
+    )
+
+    content = JSON.parse(json).dig("content")
+    
+    decoded_authorized_keys = Base64.decode64(content) 
+    
+    if(decoded_authorized_keys == authorized_keys)
+      return nil
+    else
+      content_sha = JSON.parse(json).dig("sha")
+      encoded_authorized_keys = Base64.strict_encode64(authorized_keys)
+      return encoded_authorized_keys, content_sha
+    end
 
-  # Get the latest sha for the content
-  api_url = GITHUB_REST_API_URL + "/contents/files/authorized_keys.txt"
+end
 
-  json = get_query(
-    token: ENV.fetch("GITHUB_TOKEN"),
-    api_url: api_url
-  )
+def create_new_branch_commit(content_sha, encoded_authorized_keys)
 
-  content_sha = JSON.parse(json).dig("sha")
+    # Get the main branch sha
+    json = get_query(
+      token: ENV.fetch("GITHUB_TOKEN"),
+      api_url: GITHUB_REST_API_URL + "/git/ref/heads/main"
+    )
+  
+    sha = JSON.parse(json).dig("object", "sha")
+  
+    # Create a reference which is basically creating a branch
+    json = post_query(
```

Comment:
Are you doing anything with the "json = " value from the post and put queries? If not, it's best not to assign the return values at all. i.e. 

```
post_query(...)
```

Instead of 

```
json = post_query(...)
```

Reply:
Good point @digitalronin. I have removed capturing return values.